net/libfab/xcode.h

```diff
@@ -0,0 +1,69 @@
+/* -*- C -*- */
+/*
+ * Copyright (c) 2019-2020 Seagate Technology LLC and/or its Affiliates
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For any questions about this software or licensing,
+ * please email opensource@seagate.com or cortx-questions@seagate.com.
+ *
+ */
+
+
+/**
+ * @addtogroup netlibfab
+ *
+ * @{
+ */
+
+#pragma once
+
+#ifndef __MOTR_NET_LIBFAB_XCODE_H__
+#define __MOTR_NET_LIBFAB_XCODE_H__
+
+#ifdef __KERNEL__
+#  include "lib/linux_kernel/types.h"
+#else
+#  include "lib/user_space/types.h"
+#endif
+
+
+#include "xcode/xcode_attr.h"
+
+/**
+ * libfabric module binary structures
+ * -----------------------------
+ */
+
+/** TODO: Update below struct as per libfab requirement */
+struct fab_addr {
+	uint32_t        a_type;
+	uint32_t        a_protocol;
+	uint32_t        a_port;
+} M0_XCA_RECORD M0_XCA_DOMAIN(rpc);
```

Comment:
xcode is needed for structures that are passed over network or stored on persistent store. Do we need code for fab_addr? sock.c needs xcoded structures, because it is doing its own internal packing-unpacking on network.

Reply:
looks like we do not need struct, but added this as part of framework,
if not required in future then we can remove this